builder/main.py

```diff
@@ -79,6 +77,14 @@
     PROGSUFFIX=".elf"
 )
 
+if not "stm32duino" in env.subst("$PIOFRAMEWORK"):
```

Comment:
I propose to move this logic to `arduino.py` file. You can remove values from env variable. For example,
```python
for item in ("-nostartfiles", "-nostdlib"):
    if item in env['LINKFLAGS']:
        env['LINKFLAGS'].remove(item)
```